Added Thrust Curve support.

MaxMaeder opened the pull request:
Users simply pass the path of a `.rse` file they can easily download for their motor from sites like [thrustcurve.org](https://www.thrustcurve.org/) to a `ThrustCurve()` object. This will improve the accuracy of the simulation by reflecting the accurate thrust produced by motors.

As a demonstration, I included the `.rse` for the Estes E16.

marthinwurer commented:
Pycache is still in the files changed. 

Instead of quitting, throw an exception when you hit those issues. In the future those can be handled.

Also make sure that there are newlines at the end of the file.

MaxMaeder commented:
So pycache was not in the `.gitignore`, so I added it. I then deleted the artifacts.

It seems like the convention elsewhere in the code was to quit. Take for example 
```python
    def setFromEulerAngles(self, x, y, z, mode): 
        if mode == "deg":
            self.Orientation.eulerToQuaternion(degToRad(x), degToRad(y), degToRad(z))
        elif mode == "rad":
            self.Orientation.eulerToQuaternion(x, y, z)
        else:
            quit(str(self) + "setFromEulerAngles: invalid argument, use valid 'deg' or 'rad'")
```

I've added a newline, thanks.

I do appreciate the feedback, you know a lot more about this than me.

OrlandoR4 reviewed (approved):
Very well done!
